purchase_order_analytic_search/models/purchase_order.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Eficent Business and IT Consulting Services S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+
+    _inherit = "purchase.order"
+
+    @api.multi
+    @api.depends('order_line.account_analytic_id')
+    def _compute_analytic_accounts(self):
+        res = {}
+        for purchase in self:
+            res[purchase.id] = []
+            for po_line in purchase.order_line:
```

Comment:
What you did is correct but this one is shorter:
`res['purchase_id'] = purchase.mapped("order_line.account_analytic_id").ids`

Reply:
purchase.account_analytic_ids = purchase.mapped("order_line.account_analytic_id") and get rid of all the old api stuff